### PR TITLE
[CircuitCheck] Fix cleanup check, improve diagnostic

### PR DESCRIPTION
**Pull request by boschmitt**
### Description
This PR fix the checking of ancilla clean up in the CircuitCheck, and improve diagnostics of when checks fails.

### CODE REVIEW

File: utils/CircuitCheck/UnitaryBuilder.cpp [hunk 1]
```diff
@@ -34,12 +34,16 @@ LogicalResult UnitaryBuilder::build(func::FuncOp func) {
     if (auto optor = dyn_cast<quake::OperatorInterface>(op)) {
       optor.getOperatorMatrix(matrix);
       // If the operator couldn't produce a matrix, stop the walk.
-      if (matrix.empty())
+      if (matrix.empty()) {
+        optor.emitError("Couldn't produce matrix.");
```
Comment (schweitzpgi) [inline utils/CircuitCheck/UnitaryBuilder.cpp:38]: If you use `emitOpError` I believe it gives more information about which Op is the one failing in the diagnostic.

File: utils/CircuitCheck/UnitaryBuilder.cpp [hunk 2]
```diff
@@ -70,8 +74,10 @@ WalkResult UnitaryBuilder::visitExtractOp(quake::ExtractRefOp op) {
   // We need to check whether the index is a "raw" index or not.
   if (op.hasConstantIndex())
     index = op.getRawIndex();
-  else if (failed(getValueAsInt(op.getIndex(), index)))
+  else if (failed(getValueAsInt(op.getIndex(), index))) {
+    op.emitError("Failed to get index as a interger.");
```
Comment (schweitzpgi) [inline utils/CircuitCheck/UnitaryBuilder.cpp:78]:
```suggestion
    op.emitError("Failed to get index as a integer.");
```

File: utils/CircuitCheck/UnitaryBuilder.cpp (hunk 1 above)
Comment (boschmitt) [reply utils/CircuitCheck/UnitaryBuilder.cpp]: Good point. I actually didn't know about `emittOpError`! Thanks.

**Review by amccaskey — APPROVED** [review]
LGTM

**Pull request merged**